Sources/SwiftParser/Parameters.swift

```diff
@@ -160,11 +160,20 @@ extension Parser {
     let misplacedSpecifiers = parseMisplacedSpecifiers()
 
     let names = self.parseParameterNames()
-    let colon = self.consume(if: .colon)
-    let type: RawTypeSyntax?
-    if colon != nil {
-      type = self.parseType(misplacedSpecifiers: misplacedSpecifiers)
-    } else {
+    var colon = self.consume(if: .colon)
+    // try to parse the type regardless of the presence of the preceding colon
+    // because the parameter may be unnamed
+    // e.g. { [X] in } or { (:[X]) in }
+    var type: RawTypeSyntax? = self.parseType(misplacedSpecifiers: misplacedSpecifiers)
```

Comment:
I feel this a bit scary. Could you `Lookahead.canParseType()` and check if the next token is `,` or `)`, instead of checking `.missingType`.

Reply:
I wasn't aware of the existence of `Lookahead` until earlier today, sorry for that ;(